lib/common/http2_client.go

```diff
@@ -17,58 +16,43 @@ type HTTP2Client struct {
 	sync.Mutex
 
 	client http.Client
-	conn   net.Conn
 }
 
 func NewHTTP2Client(timeout, idleTimeout time.Duration, keepAlive bool) (client *HTTP2Client, err error) {
 	if keepAlive {
 		timeout, idleTimeout = 0, 0
 	}
 
-	dialer := &net.Dialer{
-		Timeout:   1 * time.Second,
-		KeepAlive: 100000 * time.Second,
-		DualStack: true,
-	}
-
-	client = &HTTP2Client{}
-
 	transport := &http.Transport{
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: true,
 		},
 		IdleConnTimeout:   idleTimeout,
 		DisableKeepAlives: !keepAlive,
-		DialContext: func(ctx context.Context, network, addr string) (conn net.Conn, err error) {
-			conn, err = dialer.DialContext(ctx, network, addr)
-
-			client.conn = conn
-			return conn, err
-		},
+		DialContext: (&net.Dialer{
+			Timeout:   1 * time.Second,
+			KeepAlive: 100000 * time.Second,
+			DualStack: true,
+		}).DialContext,
 	}
 
 	if err = http2.ConfigureTransport(transport); err != nil {
 		return
 	}
 
-	client.client = http.Client{
-		Transport: transport,
-		Timeout:   timeout,
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse // NOTE prevent redirect
+	client = &HTTP2Client{
+		client: http.Client{
+			Transport: transport,
+			Timeout:   timeout,
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse // NOTE prevent redirect
+			},
 		},
 	}
 
 	return
 }
 
-func (c *HTTP2Client) Close() {
```

Comment:
Is there reason to remove `Close()`? We can use it to close connection from client side by force in keep-alive state

Reply:
I removed it because it is not used anywhere.
Since it is not used anywhere, we cannot make sure any implementation we provide correctly works. If we need it in the future, we can trivially re-implement it.

Reply:
@Geod24 I see, it is not used but we don't need the existing code like this method, closing connection is important for keep-alive. Please don't remove it.

Reply:
I'll remove this commit then